xml/System.IO.Pipelines/ReadResult.xml

```diff
@@ -112,9 +111,8 @@
         <ReturnType>System.Boolean</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>
-          <see langword="true" /> if the <see cref="T:System.IO.Pipelines.PipeReader" /> is complete; otherwise, <see langword="false" />.</summary>
-        <value>To be added.</value>
+        <summary>Gets a value that indicates whether the current <see cref="M:System.IO.Pipelines.PipeReader.ReadAsync(System.Threading.CancellationToken)" /> operation was completed.</summary>
```

Comment:
@anurse is right. If you call PipeReader.ReadAsync(), you won't even get ReadResult until that operation completes. This is the equivalent of Stream.ReadAsync() returning 0. It would almost be more clear to say ReadResult.IsCompleted is true if PipeWriter.Complete() was called, but that's not exactly it.

First of all, the PipeReader could be wrapping a Stream or something meaning there is no PipeWriter. And secondly, just because PipeWriter.Complete() was already called on another thread, that doesn't necessarily mean this ReadResult contains all the data that was written prior to the call to PipeWriter.Complete(). If the ReadResult doesn't contain all the data remaining in the stream, IsCompleted will always be false.

You can get multiple ReadResults with IsCompleted is true if you keep calling ReadAsync or TryRead. This can be useful if the reader is consuming only part of the buffer each time it calls ReadAsync.

Now communicating this in a clear and concise API doc summary is difficult. Maybe the summary could be as follows:

```suggestion
        <summary>Gets a value that indicates whether the end of the data stream has been reached.</summary>
```

Reply:
I like this general direction. The docs don't need to tell the user exactly which sequence of method calls generates a `true` value, they just want to know what a `true` value *means*. That wording looks great to me.